packages/web/src/services/audius-sdk/discoveryNodeSelector.ts

```diff
@@ -3,7 +3,70 @@ import { DiscoveryNodeSelectorService } from '@audius/common'
 import { env } from '../env'
 import { remoteConfigInstance } from '../remote-config/remote-config-instance'
 
+const DISCOVERY_PROVIDER_TIMESTAMP = '@audius/libs:discovery-node-timestamp'
+const CACHE_TTL = 24 * 60 * 1000 // 24 hours
+
+type CachedDiscoveryNodeTimestamp =
+  | {
+      endpoint?: string
+      timestamp?: number
+    }
+  | null
+  | undefined
+
+const getCachedDiscoveryNode = () => {
+  const cached = localStorage.getItem(DISCOVERY_PROVIDER_TIMESTAMP)
+  if (cached) {
+    try {
+      const cachedDiscoveryNodeTimestamp = JSON.parse(
+        cached
+      ) as CachedDiscoveryNodeTimestamp
+      if (
+        cachedDiscoveryNodeTimestamp &&
+        cachedDiscoveryNodeTimestamp.timestamp &&
+        cachedDiscoveryNodeTimestamp.timestamp >
+          new Date().getTime() - CACHE_TTL
+      ) {
+        const cachedDiscoveryNode =
+          cachedDiscoveryNodeTimestamp.endpoint ?? undefined
+        console.debug(
+          '[discovery-node-selector-service] Using cached discovery node',
+          cachedDiscoveryNode
+        )
+        return cachedDiscoveryNode
+      } else {
+        console.debug(
```

Comment:
Should we remove the localStorage entry in this case?

Reply:
It'll get overwritten by the listener